Clustering/Core/src/test/java/org/tribuo/neighbours/TestNeighboursPerf.java

```diff
@@ -0,0 +1,158 @@
+package org.tribuo.neighbours;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.tribuo.DataSource;
+import org.tribuo.Dataset;
+import org.tribuo.Example;
+import org.tribuo.ImmutableFeatureMap;
+import org.tribuo.MutableDataset;
+import org.tribuo.clustering.ClusterID;
+import org.tribuo.clustering.example.GaussianClusterDataSource;
+import org.tribuo.math.distance.DistanceType;
+import org.tribuo.math.la.DenseVector;
+import org.tribuo.math.la.SGDVector;
+import org.tribuo.math.neighbour.NeighboursQuery;
+import org.tribuo.math.neighbour.bruteforce.NeighboursBruteForce;
+import org.tribuo.math.neighbour.bruteforce.NeighboursBruteForceFactory;
+import org.tribuo.math.neighbour.bruteforce.NeighboursBruteForceNaive;
+import org.tribuo.math.neighbour.bruteforce.NeighboursBruteForceNaiveFactory;
+import org.tribuo.util.Util;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+public class TestNeighboursPerf {
```

Comment:
This test class is in Clustering Core module at this point to leverage `GaussianClusterDataSource`